model-optimizer/extensions/middle/InsertSelect.py

```diff
@@ -36,38 +48,63 @@ def run_before(self):
         return [ReplaceSpliceNodePattern]
 
     @staticmethod
-    def pattern():
-        return dict(
-            nodes=[('op', dict(op='Assign'))],
-            edges=[])
+    def calculate_frame_time(graph: Graph):
+        # there are either one or two inputs in Kaldi. Only main input can change delay in network.
+        # Usually ivector input has name 'ivector'.
+        inputs = graph.get_op_nodes(op='Parameter')
+        if len(inputs) == 1:
+            inp_name = inputs[0].name
+        elif len(inputs) == 2:
+            if inputs[0].name == 'ivector':
+                inp_name = inputs[1].name
+            elif inputs[1].name == 'ivector':
+                inp_name = inputs[0].name
+            else:
+                raise Error("There are 2 inputs for Kaldi model but we can't find out which one is ivector. " +
+                            "Use name \'ivector\' for according input")
+        else:
+            raise Error("There are {} inputs for Kaldi model but we expect only 1 or 2".format(len(inputs)))
```

Comment:
This transformation is executed at the middle stage for a model from any framework. May be we should add a condition for the transformation to work for Kaldi models only?